metrics/traefik.go

```diff
@@ -76,146 +87,164 @@ func NewTraefikMetricProcessor(config []common.RulesConfig) (*TraefikMetricProce
-		return nil, fmt.Errorf("Error matching Traefik log")
+	switch logFormat {
+	case Combined:
+		parsedLog, err = parseCommonLog(entry)
+	case JSON:
+		parsedLog, err = parseJsonLog(entry)
+	default:
+		return nil, fmt.Errorf("unknown log format: %s", logFormat)
 	}
 
-	for idx, name := range ApacheCombinedLogRegex.SubexpNames() {
-		if idx == 0 {
-			continue
-		}
-		mappedMatches[name] = matches[idx]
+	if err != nil {
+		common.Log.WithFields(log.Fields{
+			"entry": entry.Log,
+		}).WithError(err).Error("could not parse Traefik log")
+		return nil, fmt.Errorf("could not parse Traefik log")
 	}
 
 	// filtering and rule processing
 	for rxp, rule := range mp.Rules {
-		if !rxp.MatchString(mappedMatches["frontend_name"]) {
+		if !rxp.MatchString(parsedLog["frontend_name"].(string)) {
 			common.Log.WithFields(log.Fields{
-				"entry":   mappedMatches,
+				"entry":   parsedLog,
 				"rule_id": rule.Id,
 			}).Debug("Frontend name doesn't match regex - skipping")
 			continue
 		}
 
-		if rule.PathRegexp != nil && !rule.PathRegexp.MatchString(mappedMatches["path"]) {
+		if rule.PathRegexp != nil && !rule.PathRegexp.MatchString(parsedLog["request_path"].(string)) {
 			common.Log.WithFields(log.Fields{
-				"entry":   mappedMatches,
+				"entry":   parsedLog,
 				"rule_id": rule.Id,
 			}).Debug("Path doesn't match regex - skipping")
 			continue
 		}
 
-		if rule.MethodRegexp != nil && !rule.MethodRegexp.MatchString(mappedMatches["method"]) {
+		if rule.MethodRegexp != nil && !rule.MethodRegexp.MatchString(parsedLog["request_method"].(string)) {
 			common.Log.WithFields(log.Fields{
-				"entry":   mappedMatches,
+				"entry":   parsedLog,
 				"rule_id": rule.Id,
 			}).Debug("Method doesn't match regex - skipping")
 			continue
 		}
 
 		if !rule.Filter(entry) {
 			common.Log.WithFields(log.Fields{
-				"entry":   mappedMatches,
+				"entry":   parsedLog,
 				"rule_id": rule.Id,
 			}).Debug("Entry below threshold (sampling) - skipping")
 			continue
 		}
 
-		values, err := mp.getMetrics(entry, mappedMatches)
-		if err != nil {
-			common.Log.WithError(err).WithFields(log.Fields{
-				"entry":   mappedMatches,
-				"rule_id": rule.Id,
-			}).Error("Error processing log")
-			continue
-		}
-
-		common.Log.WithFields(log.Fields{
-			"metrics": values,
-			"rule_id": rule.Id,
-		}).Debug("Successfully derived metrics")
-
 		tags := map[string]string{
-			"frontend_name": mappedMatches["frontend_name"],
+			"frontend_name": parsedLog["frontend_name"].(string),
+			"backend_name":  parsedLog["backend_name"].(string),
 			"host_name":     entry.Kubernetes.Host,
 			"cluster_name":  entry.KubernetesClusterName,
 			"data_center":   entry.Datacenter,
 			"rule_id":       rule.Id,
 		}
 
-		pt, err := client.NewPoint(measurement, tags, values, time.Now())
+		values := map[string]interface{}{}
+
+		var timestamp time.Time
+		original_timestamp, has := parsedLog["start_utc"]
+		if has {
+			timestamp, err = time.Parse("", original_timestamp.(string))
```

Comment:
Why do we use the empty layout here?

Reply:
Good catch: 5bcc0f6a198137939f7af97c3f3434ba04f58011